pulser/sequence.py

```diff
@@ -12,32 +12,60 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from __future__ import annotations
+
 from collections import namedtuple
-from collections.abc import Iterable
+from collections.abc import Callable, Generator, Iterable, Set
 import copy
 from functools import wraps
 from itertools import chain
 import json
+from sys import version_info
+from typing import Any, cast, NamedTuple, Optional, Tuple, Union
 import warnings
 
 import numpy as np
-
+from numpy.typing import ArrayLike
```

Comment:
Ahhh, so close! Only needs a blank line after this one (I'm really sorry for being a pain with these things)